File: quark/test/emit/pet/js/pet.js
var _qrt = require("datawire-quark-core");
var pets = require('./pets');
exports.pets = pets;



function main() {
    var evil = new pets.Cat();
    var good = new pets.Dog();
    (evil).greet();
    (good).greet();
}
exports.main = main;

main();

File: quarkc/test/emit/expected/rb/pet/lib/pet.rb
module Quark
require "quark"
def self.pet; Pet; end
module Pet
require_relative 'pets' # 0 () ()
require_relative 'pet' # 0 () ()


def self.call_main()
    self.main(::DatawireQuarkCore::List.new(ARGV))

    nil
end
def self.main(args)
    
    evil = ::Quark.pets.Cat.new()
    good = ::Quark.pets.Dog.new()
    evil.greet()
    good.greet()


    nil
end

if __FILE__ == $0 then ::Quark.pet.call_main() end

end # module Pet
end # module Quark
